src/core/unit.c

```diff
@@ -636,12 +636,14 @@ static void unit_clear_dependencies(Unit *u) {
                                 hashmap_remove(other_deps, u);
 
                         unit_add_to_gc_queue(other);
+                        other->dependency_generation++;
```

Comment:
I may be mistaken, but is this over accounting? `unit_free()` calls `unit_clear_dependencies()` Wouldn't `u` and `other` update twice for the same unit? (other could be a dependency other another)